testing/trino-benchto-benchmarks/src/test/java/io/trino/sql/planner/AbstractHiveCostBasedPlanTest.java

```diff
@@ -0,0 +1,184 @@
+import io.trino.plugin.hive.TestingHiveConnectorFactory;
+import io.trino.plugin.hive.authentication.HiveIdentity;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.PartitionWithStatistics;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.UnimplementedHiveMetastore;
+import io.trino.plugin.hive.metastore.recording.HiveMetastoreRecording;
+import io.trino.plugin.hive.metastore.recording.RecordingHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.InMemoryThriftMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreConfig;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.predicate.TupleDomain;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.hive.HiveStorageFormat.TEXTFILE;
+import static io.trino.plugin.hive.HiveTestUtils.SESSION;
+import static io.trino.plugin.hive.acid.AcidTransaction.NO_ACID_TRANSACTION;
+import static io.trino.plugin.hive.metastore.PrincipalPrivileges.NO_PRIVILEGES;
+import static io.trino.plugin.hive.metastore.StorageFormat.fromHiveStorageFormat;
+import static io.trino.plugin.hive.metastore.recording.TestRecordingHiveMetastore.createJsonCodec;
+import static io.trino.plugin.hive.util.HiveUtil.toPartitionValues;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+public abstract class AbstractHiveCostBasedPlanTest
+        extends AbstractCostBasedPlanTest
+{
+    @Override
+    protected Session.SessionBuilder getSessionBuilder()
+    {
+        return testSessionBuilder().setSchema(getSchema());
+    }
+
+    @Override
+    protected ConnectorFactory getConnectorFactory()
+    {
+        RecordingMetastoreConfig recordingConfig = new RecordingMetastoreConfig()
+                .setRecordingPath(getRecordingPath())
+                .setReplay(true);
+        try {
+            RecordingHiveMetastore metastore = new RecordingHiveMetastore(
+                    new UnimplementedHiveMetastore(),
+                    new HiveMetastoreRecording(recordingConfig, createJsonCodec()));
+
+            // We need to create in memory metastore from recording metastore because recording metastore caches
+            // metadata with request or api parameters as key. So, it does caches all the partition statistics but
+            // with sampling the api parameters changes, basically statistics for certain partitions. Hence, the api
+            // won't return any statistics.
```

Comment:
Not sure if we can change this because it seems like the purpose of Recording metastore is to record each request separately. At least that's how it is implemented right now.